pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -202,11 +199,19 @@ protected void createJwtProfile(final TokenCredentials credentials, final JWT jw
         if (subject == null) {
             throw new TechnicalException("JWT must contain a subject ('sub' claim)");
         }
-
         if (!subject.contains(CommonProfile.SEPARATOR)) {
             subject = JwtProfile.class.getName() + CommonProfile.SEPARATOR + subject;
         }
 
+        final Date expirationTime = claimSet.getExpirationTime();
+        if (expirationTime != null) {
+            final Date now = new Date();
+            if (expirationTime.before(now)) {
+                logger.error("The JWT is expired: no profile is built");
+                return;
```

Comment:
It's really hard to say which case is better, I think it depends on how user use this token. So for my project, I just catch the TechnicalException in custom filter and return 401. I think if a token has been changed (invalid), then it should not pass the authentication, as well as the expiration.